### ISSUE TITLE
deserializated list only contains one element

### ISSUE THREAD

**Issue by bognari** [FasterXML/jackson-dataformat-xml#330]
Hi, I'm trying to parse some exported wiki pages in XML format. For this I use the following code.

```
val xmlMapper = XmlMapper()
val obj export = xmlMapper.readValue(xml, WikiExport::class.java)
```

In theory, my "Export" object should contain all pages (2) that are defined within the XML. However, my page list contains only the last page.  

As far as I can see, every page entry is parsed from XML and a page object is created for it. However, for each page object, a new list is created and assigned to the export object. Thus already parsed pages are deleted.

When I delete the text attribute in the revision object, the problem no longer occurs and all pages are included in the export object.

In order to surround the problem, I first decided on this ugly solution.

```
    var pagesFixed: MutableList<Page> = ArrayList()

    @JacksonXmlElementWrapper(useWrapping = false)
    @JacksonXmlProperty(localName = "page")
    var pages: MutableList<Page>? = null
        set(value) {
            field = value
            if (value != null) {
                pagesFixed.addAll(value)
            }
        }
```

Model (Yes it is Kotlin but I can't imagine that the problem is Kotlin.):
```
@JsonIgnoreProperties(ignoreUnknown = true)
@JacksonXmlRootElement(localName = "mediawiki")
class WikiExport {
    @JacksonXmlElementWrapper(useWrapping = false)
    @JacksonXmlProperty(localName = "page")
    lateinit var pages: List<Page>
}

@JsonIgnoreProperties(ignoreUnknown = true)
class Page {
    @JacksonXmlProperty(localName = "title")
    lateinit var title: String
    @JacksonXmlProperty(localName = "id")
    lateinit var pageId: String
    @JacksonXmlProperty(localName = "revision")
    lateinit var revision: Revision
}

@JsonIgnoreProperties(ignoreUnknown = true)
class Revision {
    @JacksonXmlProperty(localName = "id")
    lateinit var revisionID: String
    @JacksonXmlProperty(localName = "text")
    lateinit var text: String
}
```

XML (minimal example):
```
<mediawiki xmlns="http://www.mediawiki.org/xml/export-0.10/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.mediawiki.org/xml/export-0.10/ http://www.mediawiki.org/xml/export-0.10.xsd" version="0.10" xml:lang="en">
    <page>
      <title>Tag:railway=platform</title>
      <ns>0</ns>
      <id>24456</id>
      <revision>
        <id>1573377</id>
        <parentid>1383735</parentid>
        <timestamp>2018-02-23T04:58:48Z</timestamp>
        <contributor>
          <username>xxxxxxxx</username>
          <id>00000000</id>
        </contributor>
        <comment>/* See also */ add platform edge</comment>
        <model>wikitext</model>
        <format>text/x-wiki</format>
        <text xml:space="preserve" bytes="4169">{{ValueDescription
   |key=railway
   |value=platform
   |image=Image:Cercanías San Vicente del Raspeig.JPG
   |description=A railway platform.
   |group=Railways
   |onNode=no
   |onWay=yes
   |onArea=yes
   |onRelation=no
   |lang=en
   |wikidata=Q325358
   |combination=
  * {{Tag|name}}
  * {{Tag|ref}}
  * {{Tag|surface}}
  * {{Tag|width}}
  * {{tag|area|yes}}
  | seeAlso     =
  * {{Tag|public_transport|platform}}
  * {{Tag|highway|platform}}
  | status      = approved
  | statuslink  = Approved feature/platform
  }}
      </text>
        <sha1>jsh15pji25crcdx0vxvk0s3tfbgphbc</sha1>
      </revision>
    </page>
    <page>
      <title>Tag:barrier=sump buster</title>
      <ns>0</ns>
      <id>94664</id>
      <revision>
        <id>1603534</id>
        <parentid>1551572</parentid>
        <timestamp>2018-04-23T16:45:35Z</timestamp>
        <contributor>
          <username>xxxxxxxx</username>
          <id>00000000</id>
        </contributor>
        <minor/>
        <comment>is -&gt; are</comment>
        <model>wikitext</model>
        <format>text/x-wiki</format>
        <text xml:space="preserve" bytes="1751">{{ValueDescription
  |key=barrier
  |value=sump_buster
  |image=File:SumpBuster.jpg
  |description=A barrier to stop cars (two tracked vehicles with less than a certain ground clearance and width between tracks).
  |group=Barriers
  |lang=en
  |onNode=yes
  |onWay=no
  |onArea=no
  |onRelation=no
  |status=In use
  |wikidata=Q7638033
  }}
  </text>
        <sha1>naeytvvkmm7m57hrugb96k3c0h6gloh</sha1>
      </revision>
    </page>
</mediawiki>
```

Versions:
```
implementation("com.fasterxml.jackson.module:jackson-module-kotlin:2.9.8")
implementation("com.fasterxml.jackson.dataformat:jackson-dataformat-xml:2.9.8")
```

**Comment by cowtowncoder** [comment]
That XML is not really minimal enough to reproduce from unit test.
As to Kotlin, I wonder `lateinit` might imply.

I think at this point I would need Java-only reproduction (unit test). But before that, you may want to see if `2.11.0` might work -- there are many fixes since 2.9.8 for XML module (as well as some for Kotlin).

**Comment by cowtowncoder** [comment]
Would need a reproduction to verify; closing but may be reopened with a test.
